fem/fespace.cpp

```diff
@@ -791,7 +791,7 @@ const Operator *FiniteElementSpace::GetElementRestriction(
 {
    // Check if we have a discontinuous space using the FE collection:
    const L2_FECollection *dg_space = dynamic_cast<const L2_FECollection*>(fec);
-   if (dg_space) { return NULL; }
+   //if (dg_space) { return NULL; }
```

Comment:
@v-dobrev , why do we have to return NULL here? I was able to run my continuous galerkin advection solver by commenting the return statement.

Reply:
This is a simple optimization -- if the element restriction is the identity, there is no need to apply it.

Reply:
If I remove the comment from the check, then this example segfaults. I suspect its because I'm not using DG here, I was hoping to impose periodicity in the domain by using one of the periodic meshes. Might there be a better way to impose periodicity?

Reply:
Oh, I see -- the issue is that periodic meshes use DG space to represent the geometry and that case is not handled in the construction of the `GeometricFactors`. This should be simple to fix in the constructor of class `GeometricFactors`.

Reply:
Let me see if I can cook something up for the fix discussed above. I'll also start the MPI version on this PR.

Reply:
This was fixed recently